packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -1,42 +1,66 @@
-import React, { ReactElement, useState } from "react";
+import React, { ReactElement, useRef, useState } from "react";
 import classnames from "classnames";
+import ReactDOM from "react-dom";
+import { usePopper } from "react-popper";
+import { useOnKeyDown } from "@jobber/hooks/useOnKeyDown";
 import styles from "./ComboboxContent.css";
 import { Icon } from "../../../Icon";
 import { ComboboxContext } from "../../ComboboxProvider";
+import { ComboboxOption } from "../../Combobox.types";
 
 interface ComboboxContentProps {
   /**
-   * options for the combobox
+   * List of selectable options to display.
    */
-  readonly options: string[];
+  readonly options: ComboboxOption[];
 
   /**
-   * onSelection handler.
+   * Callback function invoked upon the selection of an option. Provides the selected option as an argument.
    */
-  readonly onSelection: (selection: string) => void;
+  readonly onSelection: (selection: ComboboxOption) => void;
 
   /**
-   * actions
+   * Optional action button(s) to display at the bottom of the list.
    */
-  readonly children: ReactElement | ReactElement[];
+  readonly children?: ReactElement | ReactElement[];
+
+  /**
+   * Placeholder text to display in the search input. Defaults to "Search".
+   */
+  readonly searchPlaceholder?: string;
 }
 
 export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
-  const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
-  const { open } = React.useContext(ComboboxContext);
+  const {
+    searchValue,
+    setSearchValue,
+    selectedOption,
+    setSelectedOption,
+    open,
+    setOpen,
+    popperRef,
+    popperStyles,
+    attributes,
+  } = useComboboxContent();
 
   const filteredOptions = props.options.filter(option =>
-    option.toLowerCase().includes(searchValue.toLowerCase()),
+    option.label.toLowerCase().includes(searchValue.toLowerCase()),
   );
 
-  return (
-    <div className={classnames(styles.content, !open && styles.hidden)}>
+  const template = (
+    <div
+      ref={popperRef}
+      data-testid="combobox-content"
```

Comment:
For the testId here and other places we should follow our guide on naming testing ids
https://jobber.atlassian.net/wiki/spaces/TEAMATL/pages/2476933462/How+should+we+declare+test+ID+s+in+Atlantis+components